stream_exercise.py

```diff
@@ -28,40 +26,32 @@ class StreamProcessor(object):
                together: if it reaches the 10th two digit number and the
                sum has not yet reached 200, then the method will stop and
                return 10.
+        """
 
-    For example, given a stream yielding "234761640930110349378289194", the
-    process method will:
-
-            1. Read two digits at a time from the stream: "23", "47", "61", etc.
-            2. Convert these digits into a number: 23, 47, 61, etc., and  make a
-               running total of these numbers: 23 + 47 equals 70. 70 + 61 equals
-               131, etc.
-            3. For this particular stream, the running total will exceed 200 after
-               5 such additions: the `process` method should return 5.
-
-    You can see the `tests.py` file for more examples of expected outcomes.
-    """
+        count = 0  # How many two-digit numbers the `process` method has added together.
+        total = 0  # The running total of sums.
 
-    def __init__(self, stream):
-        self._stream = stream
+        # I did this before looking at the solution.  Works the same, perhaps not as
+        #   clean as the other, but thought I'd leave it as I wrote it.
 
-    def process(self):
-        """
-        TODO: Implement the `process` method, as described above.
-        
-        :return: int
-        """
+        for block in iter(lambda: self._stream.read(2), ""):
```

Comment:
This would probably work better as a `while` loop:
```
        total = 0
        count = 0

        while (total <= 200) and (count < 10):
            num = self._stream.read(2)
            if len(num) < 2:
                break
            total += int(num)
            count += 1

        return count
```